src/http/data_loader.rs

```diff
@@ -156,4 +197,178 @@ mod tests {
+      ConstValue::Object(post4),
+    ]);
+    let client = MockHttpClient {
+      request_count: Arc::new(AtomicUsize::new(0)),
+      requests: Arc::new(Mutex::new(Vec::new())),
+      response: Some(mock_response),
+    };
+
+    let loader = HttpDataLoader::new(
+      Arc::new(client.clone()),
+      Some(GroupBy::new(vec!["userId".to_string()])),
+      true,
+    );
+    let loader = loader.to_data_loader(Batch::default().delay(1));
+
+    let request1 = reqwest::Request::new(reqwest::Method::GET, "http://example.com?userId=1".parse().unwrap());
+    let request2 = reqwest::Request::new(reqwest::Method::GET, "http://example.com?userId=2".parse().unwrap());
+    let headers = BTreeSet::new();
+    let key1 = DataLoaderRequest::new(request1, headers.clone());
+    let key2 = DataLoaderRequest::new(request2, headers);
+    let future1 = loader.load_one(key1);
+    let future2 = loader.load_one(key2);
+    let response = join_all([future1, future2]).await;
+    assert!(matches!(
+      response
+        .first()
+        .unwrap()
+        .as_ref()
+        .unwrap()
+        .as_ref()
+        .unwrap()
+        .body
+        .clone(),
+      ConstValue::List(..)
+    ));
+  }
+
+  #[tokio::test]
+  async fn test_batch_object_field() {
+    let mut post1 = IndexMap::<async_graphql::Name, ConstValue>::new();
+    post1.insert(Name::new("userId"), ConstValue::String("1".into()));
+    post1.insert(Name::new("title"), ConstValue::String("abc1".into()));
+    let mut post2 = IndexMap::<async_graphql::Name, ConstValue>::new();
+    post2.insert(Name::new("userId"), ConstValue::String("2".into()));
+    post2.insert(Name::new("title"), ConstValue::String("abc2".into()));
+
+    let mock_response = ConstValue::List(vec![ConstValue::Object(post1), ConstValue::Object(post2)]);
+
+    let client = MockHttpClient {
+      request_count: Arc::new(AtomicUsize::new(0)),
+      requests: Arc::new(Mutex::new(Vec::new())),
+      response: Some(mock_response),
+    };
+
+    let loader = HttpDataLoader::new(
+      Arc::new(client.clone()),
+      Some(GroupBy::new(vec!["userId".to_string()])),
+      false,
+    );
+    let loader = loader.to_data_loader(Batch::default().delay(1));
+
+    let request1 = reqwest::Request::new(reqwest::Method::GET, "http://example.com?userId=1".parse().unwrap());
+    let request2 = reqwest::Request::new(reqwest::Method::GET, "http://example.com?userId=2".parse().unwrap());
+    let headers = BTreeSet::new();
+    let key1 = DataLoaderRequest::new(request1, headers.clone());
+    let key2 = DataLoaderRequest::new(request2, headers);
+    let future1 = loader.load_one(key1);
+    let future2 = loader.load_one(key2);
+    let response = join_all([future1, future2]).await;
+    assert!(matches!(
+      response
+        .first()
+        .unwrap()
+        .as_ref()
+        .unwrap()
+        .as_ref()
+        .unwrap()
+        .body
+        .clone(),
+      ConstValue::Object(..)
+    ));
+  }
```

Comment:
Can we write an `http_spec` using yml, instead of tests in .rs files.

Reply:
Updated `n-plus-one-list.yml` and removed the list tests in dataloader.